### PR TITLE
Approximate Location "Puck" logic updates

### PR DESCRIPTION
**Pull request by neelmistry94**
This alters the logic for how the approximate location puck is draw. 

These changes allow the puck to be seen at an appropriate size at all zoom levels by doing the following:

- Draws/creates the approximate ring before updating the ring
- Passes zoom level when accounting ring size so it is relative to the current map's zoom level

### CODE REVIEW

**Review by fabian-guerra — CHANGES_REQUESTED** [review]
Is it possible to rollback files to Objc? I don't think we need Objc++ files (`.mm`)

**Comment by neelmistry94** [comment]
> Is it possible to rollback files to Objc? I don't think we need Objc++ files (`.mm`)

I worked with @julianrex on this. I believe there is a compiler error if we leave it as objc.

File: platform/ios/src/MGLFaux3DUserLocationAnnotationView.mm [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #import "MGLFaux3DUserLocationAnnotationView.h"
 
 #import "MGLMapView.h"
```
Comment (fabian-guerra) [inline platform/ios/src/MGLFaux3DUserLocationAnnotationView.mm:3]: Please remove this import.

**Review by fabian-guerra — APPROVED** [review]
There's one minor comment. Otherwise LGTM

**Comment by fabian-guerra** [comment]
I've realized that the issue is due to the fact that in the map's private header we import a c++ header, this requires importing classes to switch to Objc++ even tho no C++ is used.

File: platform/ios/src/MGLFaux3DUserLocationAnnotationView.mm (hunk 1 above)
Comment (neelmistry94) [reply platform/ios/src/MGLFaux3DUserLocationAnnotationView.mm:3]: we moved one function to map view private which is why we included it.

Comment (fabian-guerra) [reply platform/ios/src/MGLFaux3DUserLocationAnnotationView.mm:3]: Right. That's why you don't have to import `MGLMapView.h` when you have `MGLMapView_Private.h`

Comment (neelmistry94) [reply platform/ios/src/MGLFaux3DUserLocationAnnotationView.mm:3]: gotcha, I can remove it

**Comment by captainbarbosa** [comment]
@neelmistry94 can you please add an entry to the changelog as a follow-up PR?

**Comment by fabian-guerra** [comment]
@captainbarbosa this doesn't require a changelog entry because its fixing an issue in an unreleased feature.

**Pull request merged**